presto-main/src/main/java/com/facebook/presto/execution/scheduler/SectionExecutionFactory.java

```diff
@@ -335,8 +336,8 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                     // verify execution is consistent with planner's decision on dynamic lifespan schedule
                     verify(bucketNodeMap.isDynamic() == dynamicLifespanSchedule);
 
-                    if (!bucketNodeMap.isDynamic()) {
-                        stageNodeList = ((FixedBucketNodeMap) bucketNodeMap).getBucketToNode().stream()
+                    if (!bucketNodeMap.isDynamic() || ((DynamicBucketNodeMap) bucketNodeMap).ifInitialMapExist()) {
```

Comment:
If `ifInitialMapExist` becomes an interface method, then we don't need `!bucketNodeMap.isDynamic()` anymore.